### ISSUE TITLE
Document read and write request paths

### ISSUE THREAD

**Issue by jseldess** [cockroachdb/docs#3534]
Our architecture docs cover the components involved in reads and writes, but it would be nice to have docs that put the pieces together, in greater detail and with examples. 

For example, currently, there's not a good way to understand the exact sequence of actions for a simple write, how both the raft log and kv come into play and when, and when data is actually written to disk. The [Life of a SQL query](https://github.com/cockroachdb/cockroach/blob/master/docs/tech-notes/life_of_a_query.md) tech note is probably the best current source of information. 

cc: @bdarnell, @robert-s-lee, @andreimatei.

**Comment by jseldess** [comment]
Thanks for @andreimatei for finding this: For future reference, the Raft leader and its followers write new Raft log entries to disk in parallel (i.e., the leader's write isn't blocking): https://github.com/cockroachdb/cockroach/commit/b67eb696f6a064d6a3a18517c1ed5525faa931e8

**Comment by robert-s-lee** [comment]
https://github.com/cockroachdb/docs/issues/2300 has sequence diagram that may be helpful.

**Comment by jseldess** [comment]
Ah, yes. Thank you, @robert-s-lee. I knew there was a related issue but couldn't find it.
